### PR TITLE
implement posting results from poll

### PR DESCRIPTION
**Pull request by Ipkhchan**
What this PR implments:

- when /postwinners [correct answer] is run, server looks up past messages to find the Poll message, parses the users from it and calls postLosers and postWinners to update the database.

To Test:
1. postwinners handler requires ACCESS_TOKEN (authorization to query slack api for messages in channel) and ADMIN_IDS (authorization to use post results to the database). Create a .env file in root directory of app and create ACCESS_TOKEN and ADMIN_IDS fields in the format specified at the following link: https://www.npmjs.com/package/dotenv. Add your user_id (looks like UH1GHEGUS which you can get from your profile on slack) to ADMIN_IDS. Ping me to get the ACCESS_TOKEN.
2. yarn install
3. yarn start
4. Use postman (or similar) to make a post request to localhost:3000/postwinners with the following fields and values in body of request (make sure it is in x-www-form-urlencoded):

{
"text": "3 months",
"user_id": "[YOUR USER ID, OR WHOEVER YOU AUTHORIZED IN .ENV FILE]"
}

("3 months" is the correct answer to the poll in the #test channel in our test workspace)

5. Check database for changes to documents UH1GHEGUS and UHPKWDQJK.

### CODE REVIEW

File: middleware/handleGet.js [hunk 1]
```diff
@@ -6,17 +6,19 @@ const {
...
-const orderedColumns = require('../constants');
+const {orderedColumns, commands} = require('../constants');
 
 async function handleGet(ctx) {
   const {text} = ctx.request.body;
 
-  let [category, numUsers] = text.split(' ');
-  numUsers = parseInt(numUsers);
+  const command = commands.find((command) => text.includes(command)) || null;
+  const numUsers = Number(text.match(/\d+/));
```
Comment (joeyzhoucode) [inline middleware/handleGet.js]: This finds the first sequence of numbers in the text. Do we want that? Something like `"3 wins 4 losses 5"` would return `wins` as command and `3` as number of users. I think we can make the regex stricter by only accepting the `${command} ${numUsers}` format and throw an error if it's not in the format we want, but that depends on how much time we want to spend on this hackdays project.

File: middleware/handlePost.js [hunk 2]
```diff
@@ -1,8 +1,98 @@
...
 const {postWinners, postLosers} = require('./crud-functions/index.js');
 
 async function handlePost(ctx) {
-  const {channel_id, text} = ctx.request.body;
-  postWinners(text);
+  const {user_id, response_url, text: correctAnswer} = ctx.request.body;
+
+  const adminIds = process.env.ADMIN_IDS.split(',');
+  if (!adminIds.includes(user_id)) {
+    ctx.status = 200;
```
Comment (joeyzhoucode) [inline middleware/handlePost.js]: Shouldn't this return 401?

File: middleware/handlePost.js [hunk 3]
```diff
@@ -1,8 +1,98 @@
...
+    ctx.body = 'You are not authorized to post results';
+    return;
+  }
+
+  if (!correctAnswer) {
+    ctx.status = 200;
```
Comment (joeyzhoucode) [inline middleware/handlePost.js]: I think this should return 422.

**Review by joeyzhoucode — APPROVED** [review]
LGTM

File: middleware/handlePost.js (hunk 2 above)
Comment (Ipkhchan) [reply middleware/handlePost.js]: If we return 401, Slack just displays a "generic" error message. I'd rather return successfully with a message that the Slack user can easily understand. 

Screen Shot 2019-04-07 at 11 47 36 AM

File: middleware/handleGet.js (hunk 1 above)
Comment (Ipkhchan) [reply middleware/handleGet.js]: good point, I'll change it back.

**Pull request merged**